include/openthread/dnssd_server.h

```diff
@@ -159,8 +169,11 @@ typedef struct otDnssdCounters
     uint32_t mNameErrorResponse;      ///< The number of name error responses
     uint32_t mNotImplementedResponse; ///< The number of 'not implemented' responses
     uint32_t mOtherResponse;          ///< The number of other responses
+    uint32_t mResolvedBySrp;          ///< The number of queries completely resolved by the local SRP server
+    uint32_t mTotalFailure;           ///< Sum of all failures.
```

Comment:
Why this filed is in otDnssdCounters, while other fields are not?

Consider

```suggestion
    uint32_t mResolvedByUpstream;
    uint32_t mUpstreamFailures;           ///< Sum of all failures.
```

(Then response count should be mResolvedByUpstream - mUpstreamFailures)

Reply:
Agree. This counter is counting failures of "DNS upstream mechasnim" only (`mTotalFailures` make it sound as if it is associated with all failures on DNS-SD, e.g. when resolved by SRP, etc).

Reply:
Got it, thanks you two!! Will update in next commit.